src/query/sql/src/planner/plans/join.rs

```diff
@@ -437,10 +443,13 @@ impl Operator for Join {
                     + f64::max(right_cardinality, inner_join_cardinality)
                     - inner_join_cardinality
             }
-            JoinType::LeftSemi | JoinType::LeftAnti | JoinType::LeftMark | JoinType::Single => {
+            JoinType::LeftSemi | JoinType::LeftAnti | JoinType::LeftMark | JoinType::LeftSingle => {
                 left_cardinality
             }
-            JoinType::RightSemi | JoinType::RightAnti | JoinType::RightMark => right_cardinality,
+            JoinType::RightSemi
+            | JoinType::RightAnti
+            | JoinType::RightMark
+            | JoinType::RightSingle => right_cardinality,
```

Comment:
Why put `RightSingle`  here?

Reply:
If the join type is right single, HashJoin's output row num is right_cardinality.